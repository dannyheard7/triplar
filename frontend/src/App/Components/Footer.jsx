import React from "react";

export default function Footer(props) {
    return (
        <footer className="footer">
            Made By Danny Heard
            triplar7@gmail.com
        </footer>
    );
}
