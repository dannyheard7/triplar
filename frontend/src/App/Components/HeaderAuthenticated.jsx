import React from "react";
import {Link} from 'react-router-dom';

import {unsetUser} from "../../Auth/utils/actions";
import {withRouter} from "react-router";
import {connect} from "react-redux";

export class HeaderAuthenticated extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            showDropdown: false
        };

        this.onLogoutClick = this.onLogoutClick.bind(this);
        this.showMenu = this.showMenu.bind(this);
        this.closeMenu = this.closeMenu.bind(this);
    }

    onLogoutClick(e) {
        this.props.dispatch(unsetUser());
    }

    showMenu(event) {
        event.preventDefault();

        this.setState({showDropdown: true}, () => {
            document.addEventListener('click', this.closeMenu);
        });
    }

    closeMenu() {
        this.setState({showDropdown: false}, () => {
            document.removeEventListener('click', this.closeMenu);
        });
    }

    componentWillUnmount() {
        document.removeEventListener('click', this.closeMenu);
    }

    render() {
        const {user} = this.props.auth;
        
        return (
            <div className="header">
                <Link to='/' activeclassname="active" className="brand">Triplar</Link>

                <button className="navbar-toggle" type="button" data-toggle="collapse"
                        data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
                        aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <nav className="navbar">
                    <span className="nav-item nav-dropdown">
                        <button className="dropdown-toggle" onClick={this.showMenu}>
                            {user.email}
                            <i className="fa fa-caret-down"></i>
                            {this.state.showDropdown &&
                            <div className="dropdown-content" id="user-dropdown">
                                <a className="dropdown-item" id="logout-button" onClick={this.onLogoutClick}>Logout</a>
                            </div>
                            }
                        </button>
                    </span>
                </nav>
            </div>
        );
    }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

// make Redux state piece of `login` and our action `loginRequest`
// available in this.props within our component
export default withRouter(connect(mapStateToProps)(HeaderAuthenticated));