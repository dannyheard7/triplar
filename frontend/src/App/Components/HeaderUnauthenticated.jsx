import React from "react";
import {Link} from 'react-router-dom';

export default function HeaderUnauthenticated(props) {
    return (
        <div className="header">
            <Link to='/' activeclassname="active" className="brand">Triplar</Link>

            <button className="navbar-toggle" type="button" data-toggle="collapse"
                    data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
                    aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>

            <nav className="navbar">
                <span className="nav-item" id="login-nav-item">
                    <Link to='/login' className="nav-link">Login</Link>
                </span>
                <span className="nav-item" id="register-nav-item">
                    <Link to='/register' className="nav-link">Register</Link>
                </span>
            </nav>
        </div>
    );
}
