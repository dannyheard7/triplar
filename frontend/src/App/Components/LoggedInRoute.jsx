import React from "react";
import {connect} from "react-redux";
import {Redirect, Route, withRouter} from "react-router-dom";

export class LoggedInRoute extends React.Component {

    render() {
        const {isAuthenticated, Component, ...rest} = this.props;

        if (isAuthenticated) {
            return (
                <Route {...rest} Component/>
            );
        } else {
            return (
                <Redirect to={{pathname: '/login', state: {from: this.props.location}}}/>
            );
        }
    }
}

function mapStateToProps(state, props) {
    return {
        isAuthenticated: state.auth.isAuthenticated
    }
}

export default withRouter(connect(mapStateToProps)(LoggedInRoute))