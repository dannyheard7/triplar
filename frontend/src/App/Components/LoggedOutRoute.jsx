import React from "react";
import {connect} from "react-redux";
import {Redirect, Route, withRouter} from "react-router-dom";

/* Only routes logged out users can see */
export class LoggedOutRoute extends React.Component {

    render() {
        const {isAuthenticated, Component, ...rest} = this.props;

        if (!isAuthenticated) {
            return (
                <Route {...rest} Component/>
            );
        } else {
            return (
                <Redirect to="/trips" />
            );
        }
    }
}

function mapStateToProps(state, props) {
    return {
        isAuthenticated: state.auth.isAuthenticated
    }
}

export default withRouter(connect(mapStateToProps)(LoggedOutRoute))