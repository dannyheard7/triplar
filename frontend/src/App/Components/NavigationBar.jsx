import React from "react";
import {connect} from "react-redux";
import {Link} from 'react-router-dom';

import {logout} from "Auth/actions";

export class NavigationBar extends React.Component {

    constructor(props){
        super(props);

        this.onLogoutClick = this.onLogoutClick.bind(this);
    }

    onLogoutClick(e) {
        this.props.dispatch(logout());
    }

    render() {
        const {isAuthenticated, user} = this.props;

        return (
            <nav className="navbar navbar-expand-lg navbar-light bg-light">
                {isAuthenticated &&
                    <Link to='/board' activeclassname="active" className="navbar-brand">Triplar</Link>
                }
                {!isAuthenticated &&
                    <Link to='/login' activeclassname="active" className="navbar-brand">Triplar</Link>
                }

                <button className="navbar-toggler" type="button" data-toggle="collapse"
                    data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
                    aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>



                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    {!isAuthenticated &&
                        <ul className="navbar-nav ml-auto">
                            <li className="nav-item" id="login-nav-item">
                                <Link to='/login' className="nav-link">Login</Link>
                            </li>
                            <li className="nav-item" id="register-nav-item">
                                <Link to='/register' className="nav-link">Register</Link>
                            </li>
                        </ul>
                    }
                    {isAuthenticated &&
                         <ul className="navbar-nav ml-auto">
                             <li className="nav-item dropdown">
                                 <a className="nav-link dropdown-toggle" href="#" id="user-dropdown" role="button"
                                    data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" className="nav-link dropdown-toggle">
                                     {user.email}
                                 </a>
                                 <div className="dropdown-menu" aria-labelledby="user-dropdown">
                                     <a className="dropdown-item" id="logout-button" onClick={this.onLogoutClick}>Logout</a>
                                 </div>
                             </li>
                        </ul>
                    }
                </div>

            </nav>
        );
    }
}

function mapStateToProps(state) {
    return {
        isAuthenticated: state.auth.isAuthenticated,
        user: state.auth.user
    }
}

export default connect(mapStateToProps)(NavigationBar)