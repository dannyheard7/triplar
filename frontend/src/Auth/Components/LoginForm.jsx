import React from "react";

import FormFieldGroup from "Forms/Components/FormFieldGroup";
import NonFieldErrors from "Forms/Components/NonFieldErrors";

export default class LoginForm extends React.Component {
    render() {
        return (
            <form onSubmit={this.props.onSubmit}>
                <NonFieldErrors errors={this.props.errors.non_field_errors}/>
                <FormFieldGroup errors={this.props.errors.email} label="Email" name="email" type="email"
                                onChange={this.props.onChange} required="true"/>
                <FormFieldGroup errors={this.props.errors.password} label="Password" name="password"
                                type="password" onChange={this.props.onChange} required="true"/>
                <button type="submit" className="btn btn-primary" id="login-submit">Login</button>
            </form>
        );
    }
}