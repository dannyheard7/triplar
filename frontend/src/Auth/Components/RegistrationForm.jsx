import React from "react";

import FormFieldGroup from "Forms/Components/FormFieldGroup";
import NonFieldErrors from "Forms/Components/NonFieldErrors";


export default class RegistrationForm extends React.Component {
    render() {
        return (
            <form onSubmit={this.props.onSubmit}>
                {this.props.errors && <NonFieldErrors errors={this.props.errors.non_field_errors} />}
                <FormFieldGroup errors={this.props.errors.email} label="Email" name="email" type="email"
                                required={true}/>
                <FormFieldGroup errors={this.props.errors.password} label="Password" name="password"
                                type="password"required={true}/>
                <FormFieldGroup errors={this.props.errors.firstName} label="First Name" name="firstName"
                                type="text" required={true}/>
                <FormFieldGroup errors={this.props.errors.lastName} label="Last Name" name="lastName" type="text"
                                required={true}/>
                <button type="submit" className="btn btn-primary">Submit</button>
            </form>
        );
    }
}
