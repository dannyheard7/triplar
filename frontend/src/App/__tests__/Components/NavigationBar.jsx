import React from "react";
import {shallow} from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';

import {NavigationBar} from "App/Components/NavigationBar";

const actions = require('Auth/actions');
const faker = require('faker');
const renderer = new ShallowRenderer();


describe('<NavigationBar />', () => {

    describe('when authenticated', () => {
        const user = {'email': "test@example.com"};

        const props = {
            isAuthenticated: true,
            user: user,
            dispatch: jest.fn(),
            history: {push: jest.fn()}
        };

        const logoutSpy = jest.spyOn(actions, 'logout');

        afterEach(function () {
            logoutSpy.mockReset();
            props.dispatch.mockReset();
        });


        test('calls onLogoutClick when logout button is clicked', () => {
            const logoutClickStub  = jest.spyOn(NavigationBar.prototype, "onLogoutClick");

            const container = shallow(<NavigationBar  {...props}/>);
            container.find("#logout-button").simulate('click');
            expect(logoutClickStub).toBeCalled();
        });

        test('displays user email', () => {
            const container = shallow(<NavigationBar {...props}/>);
            expect(container.find("#user-dropdown").text()).toEqual(user['email'])
        });

        test('dispatches logout on logoutClick', () => {
            const container = shallow(<NavigationBar {...props}/>);

            const mock = jest.fn();
            logoutSpy.mockReturnValue(mock);

            container.instance().onLogoutClick();

            expect(logoutSpy).toBeCalled();
            expect(props.dispatch).toBeCalledWith(mock);
        });

        test('redirects to logout on logoutClick', () => {
            const container = shallow(<NavigationBar {...props}/>);

            const mock = jest.fn();
            logoutSpy.mockReturnValue(mock);

            container.instance().onLogoutClick();

            expect(logoutSpy).toBeCalled();
            expect(props.history.push).toBeCalledWith('/login');

        });

        test('renders correctly', () => {
            const result = renderer.render(<NavigationBar  />);
            expect(result).toMatchSnapshot();
        });

    });

    describe('when not authenticated', () => {
        const props = {
            isAuthenticated: false,
        };

        test('renders correctly', () => {
            const result = renderer.render(<NavigationBar  />);
            expect(result).toMatchSnapshot();
        });
    });
});