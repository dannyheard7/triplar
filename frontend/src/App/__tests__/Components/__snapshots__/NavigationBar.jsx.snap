// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> when authenticated renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <Link
    activeclassname="active"
    className="navbar-brand"
    replace={false}
    to="/login"
  >
    Triplar
  </Link>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
        id="login-nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/login"
        >
          Login
        </Link>
      </li>
      <li
        className="nav-item"
        id="register-nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/register"
        >
          Register
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`<NavigationBar /> when not authenticated renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <Link
    activeclassname="active"
    className="navbar-brand"
    replace={false}
    to="/login"
  >
    Triplar
  </Link>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
        id="login-nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/login"
        >
          Login
        </Link>
      </li>
      <li
        className="nav-item"
        id="register-nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/register"
        >
          Register
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
